### PR TITLE
Tests: Extend InternalStatsTests

### PR DESCRIPTION
**Pull request by cbuescher**
Currently we don't test for count = 0 which will make a difference when adding
tests for parsing for the high level rest client. Also min/max/sum should also
be tested with negative values and on a larger range and we could use a 
randomized numeric format.

### CODE REVIEW

**Comment by cbuescher** [comment]
@colings86 I pulled this out of a current WIP Pr, can you have a quick look if this makes sense to you?

File: core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalStatsTests.java
Comment (colings86) [inline core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalStatsTests.java]: Might be worth using `frequently()` here so the `0` case has more of a chance of being tested (since its more likely to cause issues than a random positive number). Also not sure why we only went up to 50 before, seems to me like we should be testing much larger ranges (probably worth going above `Integer.Max_VALUE`) so we make sure that we are ok for very high values since we expect aggregations to collect a lot of documents

File: core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalStatsTests.java
Comment (colings86) [inline core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalStatsTests.java]: I think these ranges are also too small, we should probably test a large range of values since this aggregation will likely be used for large ranges. Mostly here I'm thinking this might catch bugs in the reduce logic rather than the serialisation logic.

**Review by colings86 — APPROVED** [review]
LGTM but I left two small comments, feel free to push without another review

File: core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalStatsTests.java
Comment (cbuescher) [inline core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalStatsTests.java]: I increased the range already but this leads to rounding errors in the reduce logic so we need to be more lenient with the delta here then: https://github.com/elastic/elasticsearch/pull/24212/files/ac7bfcc469014fb376fbbbd7396fc726d55450c8#diff-3d8a3c85fc5c06b0859a2a4749fb834dL61

File: core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalStatsTests.java
Comment (cbuescher) [inline core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalStatsTests.java]: So would still use a positive value more frequently though. But I will change this and see if it breaks anything.

**Comment by cbuescher** [comment]
@colings86 thanks for the review, I made a few slight changes according to your comments and will wait for CI to pass

File: core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalStatsTests.java
Comment (colings86) [inline core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalStatsTests.java]: I actually meant to keep this with a range of random value for the frequently branch. Something like 
```
long count = frequently() ? randomPositiveLong() : 0;
```

Comment (cbuescher) [inline core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalStatsTests.java]: Oh, my mistake, thats what I also intended. I was a bit to fast here, will change that.

Comment (cbuescher) [inline core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalStatsTests.java]: randomPositiveLong() won't work though I think because then we run the risk of overflowing long when summing those values. Interger.MAX_VALUE should work I think.

**Pull request merged**
